File: scripts/cleanup-out.mjs
#!/usr/bin/env node
/**
 * Post-build cleanup: remove unnecessary files from Next.js static export
 * to stay under Cloudflare Pages 20,000 file limit.
 */
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const outDir = path.resolve(process.argv[2] || path.join(__dirname, "..", "out"));

function countFiles(dir) {
  let count = 0;
  for (const entry of fs.readdirSync(dir, { withFileTypes: true })) {
    if (entry.isDirectory()) {
      count += countFiles(path.join(dir, entry.name));
    } else {
      count++;
    }
  }
  return count;
}

// Count before
const before = countFiles(outDir);
console.log(`Before cleanup: ${before} files`);

// Delete RSC .txt files, but preserve ads.txt
let txtCount = 0;
function removeTxt(dir) {
  for (const entry of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      removeTxt(full);
    } else if (entry.name.endsWith(".txt") && entry.name !== "ads.txt") {
      fs.unlinkSync(full);
      txtCount++;
    }
  }
}
removeTxt(outDir);
console.log(`Deleted .txt files: ${txtCount}`);

// Count after
const after = countFiles(outDir);
console.log(`After cleanup: ${after} files`);
console.log(`Saved: ${before - after} files`);

File: scripts/cleanup-out.py
"""
Post-build cleanup: remove unnecessary files from Next.js static export
to stay under Cloudflare Pages 20,000 file limit.

Deletes:
1. logos/ directory (1,042 local favicon files, logos are served via Google Favicons URL)
2. All .txt RSC payload files (Next.js Flight protocol, not needed for static site rendering)
"""
import os, shutil, sys

out_dir = os.path.abspath(sys.argv[1] if len(sys.argv) > 1 else "out")

# 1. Count initial files
total_before = sum(len(files) for _, _, files in os.walk(out_dir))
print(f"Before cleanup: {total_before} files")

# 2. Delete logos/ directory
logos_dir = os.path.join(out_dir, "logos")
if os.path.isdir(logos_dir):
    count = sum(len(files) for _, _, files in os.walk(logos_dir))
    shutil.rmtree(logos_dir)
    print(f"Deleted logos/: {count} files")

# 3. Delete all .txt files (RSC payloads)
txt_count = 0
for root, dirs, files in os.walk(out_dir):
    for f in files:
        if f.endswith(".txt"):
            os.remove(os.path.join(root, f))
            txt_count += 1
print(f"Deleted .txt files: {txt_count}")

# 4. Count final
total_after = sum(len(files) for _, _, files in os.walk(out_dir))
print(f"After cleanup: {total_after} files")
print(f"Saved: {total_before - total_after} files")
